src/controllers/product.controller.js

```diff
@@ -45,25 +50,43 @@ export const getAllProducts = async (req, res) => {
 };
 export const removeExpiredProducts = async (req, res) => {
   try {
-    const {isExpired} = req.body;
+    const { isExpired } = req.body;
     const productId = req.params.productId;
     const isAvailable = false;
 
-    await ProductService.updateProduct({isExpired,isAvailable },productId );
-    sendEmail(emailConfig({email:req.user.email,subject:'Product has expired', content:expiredProductMessage(req.product)}));
+    await ProductService.updateProduct({ isExpired, isAvailable }, productId);
+    sendEmail(
+      emailConfig({
+        email: req.user.email,
+        subject: 'Product has expired',
+        content: expiredProductMessage(req.product),
+      })
+    );
 
-    return res.status(200).json({message:'Expired product was removed successfully'});
-  } catch (error) {
     return res
-      .status(500)
-      .json({
-        error: error.message,
-        message: 'Failed to remove expired product from list',
-      });
+      .status(200)
+      .json({ message: 'Expired product was removed successfully' });
+  } catch (error) {
+    return res.status(500).json({
+      error: error.message,
+      message: 'Failed to remove expired product from list',
+    });
   }
 };
 
-const repetitionDuration =process.env.CRON_PERIOD? durationToCronRepetition(process.env.CRON_PERIOD): knownSchedulingTime.everySecond;
-schedule(repetitionDuration,function () {
+const repetitionDuration = process.env.CRON_PERIOD
+  ? durationToCronRepetition(process.env.CRON_PERIOD)
+  : knownSchedulingTime.everySecond;
+schedule(repetitionDuration, function () {
   delistExpiredProducts();
 });
+export const getSpecificProduct = async (req, res) => {
+  const { role, id } = req.user;
+  let product=req.product
+    if (role === 'seller'&&product.userId != id) {
+        return res
+          .status(404)
+          .json({ message: "Product doesn't exist in your collection" });
+    }
```

Comment:
check if product belongs to the seller.

Reply:
In additional, this should be a middleware